invokeai/backend/stable_diffusion/diffusers_pipeline.py

```diff
@@ -193,6 +193,15 @@ def _adjust_memory_efficient_attention(self, latents: torch.Tensor):
             else:
                 raise Exception("torch-sdp attention slicing not available")
 
+        # See https://github.com/invoke-ai/InvokeAI/issues/7049 for context.
+        # Bumping torch from 2.2.2 to 2.4.1 caused the sliced attention implementation to produce incorrect results.
+        # For now, if a user is on an MPS device and has not explicitly set the attention_type, then we select the
+        # non-sliced torch-sdp implementation. This keeps things working on MPS at the cost of increased peak memory
+        # utilization.
+        if torch.backends.mps.is_available():
+            assert hasattr(torch.nn.functional, "scaled_dot_product_attention")
```

Comment:
Should we do this as an if else like the statement above this rather than using assert?

Reply:
The checks are totally unnecessary given that we pin the torch version. I'll just rip them all out.